src/main/scala/cromwell/engine/backend/BackendCall.scala

```diff
@@ -54,21 +54,37 @@ trait ExecutionHandle {
   def result: ExecutionResult
 }
 
+sealed trait ReturnCode {
+  def code: Int
+
+  def asScriptReturnCode = this match {
+    case script: ScriptReturnCode => Option(code)
+    case _ => None
+  }
+
+  def asBackendReturnCode = this match {
+    case script: BackendReturnCode => Option(code)
+    case _ => None
+  }
```

Comment:
commenting purely on @Horneth 's last statement about the trait knowing about implementors and not about this particular bit, FWIW it's viewed as the more functional approach of ADTs to have the trait encode most/all of the logic on behalf of its implementors vs the more OO approach where the trait is the lightweight bit. I don't generally have a strong opinion either way, sometimes I think it looks nicer one way, sometimes the other.